### PR TITLE
update Windows ARM build requirements

### LINKED ISSUE #50661: Can't build runtime in Windows arm64
I'm unable to build runtime in my Surface Pro X, which has Windows 10 ARM64. The error says the file `jitinterface_arm64.dll` could not be found.

I double checked that I had all the requirements installed. I also made sure to install CMAKE using the VS Installer, as described [here](https://devblogs.microsoft.com/cppblog/windows-arm64-support-for-cmake-projects-in-visual-studio/), since kitware / cmake.org do not offer a Windows installer for ARM64.

I also tried looking for the file in the File Explorer from the root of the runtime directory, in case it was being misplaced, but I couldn't find it.

I also tried again after downloading both the Windows ARM64 SDK installers for .NET 5.0 and for .NET 6.0 (latest) and made sure they were placed in `C:\Program Files\dotnet`, which is where `$PATH` expects them in this computer. But that didn't help.

Below is the command I used, the output, and I also attached the generated binlog.

Am I missing something, or is building in ARM64 Windows still unsupported?

[msbuild.zip](https://github.com/dotnet/runtime/files/6251263/msbuild.zip)

<details>
  <summary>C:\Users\calope\source\repos\runtime> .\build.cmd clr+libs -rc release -arch arm64 /bl</summary>

  ```
  Determining projects to restore...
  Tool 'coverlet.console' (version '1.7.2') was restored. Available commands: coverlet
  Tool 'dotnet-reportgenerator-globaltool' (version '4.5.8') was restored. Available commands: reportgenerator
  Tool 'microsoft.dotnet.xharness.cli' (version '1.0.0-prerelease.21180.1') was restored. Available commands: xharness
  Tool 'microsoft.visualstudio.slngen.tool' (version '5.0.5') was restored. Available commands: slngen

  Restore was successful.
  All projects are up-to-date for restore.
  Determining projects to restore...
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.DirectoryServices\src\System.DirectoryServices.csproj (in 263 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Diagnostics.PerformanceCounter\src\System.Diagnostics.PerformanceCounter.csproj (in 239 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Diagnostics.Process\src\System.Diagnostics.Process.csproj (in 262 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.DirectoryServices.AccountManagement\src\System.DirectoryServices.AccountManagement.csproj (in 278 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Diagnostics.EventLog\ref\System.Diagnostics.EventLog.csproj (in 221 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Diagnostics.EventLog\src\System.Diagnostics.EventLog.csproj (in 250 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.DirectoryServices\ref\System.DirectoryServices.csproj (in 260 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.CodeDom\ref\System.CodeDom.csproj (in 287 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Data.OleDb\src\System.Data.OleDb.csproj (in 27 ms).  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Configuration.ConfigurationManager\ref\System.Configuration.ConfigurationManager.csproj (in 27 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Data.OleDb\ref\System.Data.OleDb.csproj (in 64 ms).  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Management\src\System.Management.csproj (in 46 ms).  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Configuration.ConfigurationManager\src\System.Configuration.ConfigurationManager.csproj (in 95 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Management\ref\System.Management.csproj (in 32 ms).  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.IO.Ports\src\System.IO.Ports.csproj (in 48 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Net.Http.Json\ref\System.Net.Http.Json.csproj (in 449 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Net.Http.Json\src\System.Net.Http.Json.csproj (in 456 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.ServiceProcess.ServiceController\ref\System.ServiceProcess.ServiceController.csproj (in 22 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Win32.Registry.AccessControl\src\Microsoft.Win32.Registry.AccessControl.csproj (in 31 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.VisualBasic.Core\src\Microsoft.VisualBasic.Core.vbproj (in 30 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Win32.Registry\src\Microsoft.Win32.Registry.csproj (in 57 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Speech\src\System.Speech.csproj (in 106 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.ServiceProcess.ServiceController\src\System.ServiceProcess.ServiceController.csproj (in 113 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Runtime.Caching\src\System.Runtime.Caching.csproj (in 31 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Formats.Cbor\src\System.Formats.Cbor.csproj (in 22 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Formats.Cbor\ref\System.Formats.Cbor.csproj (in 15 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Logging.EventLog\src\Microsoft.Extensions.Logging.EventLog.csproj (in 57 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Logging\src\Microsoft.Extensions.Logging.csproj (in 62 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Logging.TraceSource\src\Microsoft.Extensions.Logging.TraceSource.csproj (in 80 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Diagnostics.DiagnosticSource\src\System.Diagnostics.DiagnosticSource.csproj (in 850 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\externals.csproj (in 242 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Logging.EventSource\src\Microsoft.Extensions.Logging.EventSource.csproj (in 84 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Logging.Debug\src\Microsoft.Extensions.Logging.Debug.csproj (in 75 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Logging.Configuration\src\Microsoft.Extensions.Logging.Configuration.csproj (in 65 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Logging.Console\src\Microsoft.Extensions.Logging.Console.csproj (in 100 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Http\src\Microsoft.Extensions.Http.csproj (in 86 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Hosting.Systemd\src\Microsoft.Extensions.Hosting.Systemd.csproj (in 57 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Hosting.WindowsServices\ref\Microsoft.Extensions.Hosting.WindowsServices.csproj (in 94 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Security.Permissions\ref\System.Security.Permissions.csproj (in 78 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\System.Security.Permissions\src\System.Security.Permissions.csproj (in 89 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Hosting\src\Microsoft.Extensions.Hosting.csproj (in 171 ms).
  Restored C:\Users\calope\source\repos\runtime\src\libraries\Microsoft.Extensions.Hosting.WindowsServices\src\Microsoft.Extensions.Hosting.WindowsServices.csproj (in 168 ms).
  461 of 503 projects are up-to-date for restore.
  System.Private.CoreLib.Generators -> C:\Users\calope\source\repos\runtime\artifacts\bin\System.Private.CoreLib.Generators\netstandard2.0-Release\System.Private.CoreLib.Generators.dll
  System.Private.CoreLib -> C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\IL\System.Private.CoreLib.dll
  DIALib -> C:\Users\calope\source\repos\runtime\artifacts\bin\DIALib\Release\DIALib.dll
  DacTableGen -> C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\DacTableGen\DacTableGen.dll
  runtime ->
  BUILD: Starting Build at 14:00:45.28
  BUILD: Commencing CoreCLR product build
  BUILD: Checking prerequisites
  Program 'cmake.exe' failed to run: The specified executable is not a valid application for this OS platform.At C:\Users\calope\source\repos\runtime\eng\native\set-cmake-path.ps1:36 char:25
  +   $version = [Version]$(& $cmakePath --version | Select-String -Patte ...
  +                         ~~~~~~~~~~~~~~~~~~~~~~.
  runtime ->
  BUILD: Starting Build at 14:00:53.18
  BUILD: Commencing CoreCLR product build
  BUILD: Checking prerequisites
  Program 'cmake.exe' failed to run: The specified executable is not a valid application for this OS platform.At C:\Users\calope\source\repos\runtime\eng\native\set-cmake-path.ps1:36 char:25
  +   $version = [Version]$(& $cmakePath --version | Select-String -Patte ...
  +                         ~~~~~~~~~~~~~~~~~~~~~~.
  runincontext -> C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\runincontext.dll
  ILCompiler.Diagnostics -> C:\Users\calope\source\repos\runtime\artifacts\bin\ILCompiler.Diagnostics\arm64\Release\ILCompiler.Diagnostics.dll
  ILCompiler.Reflection.ReadyToRun -> C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\ILCompiler.Reflection.ReadyToRun.dll
  R2RDump -> C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\R2RDump\R2RDump.dll
  ILCompiler.TypeSystem.ReadyToRun -> C:\Users\calope\source\repos\runtime\artifacts\bin\ILCompiler.TypeSystem.ReadyToRun\arm64\Release\ILCompiler.TypeSystem.ReadyToRun.dll
  dotnet-pgo -> C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\dotnet-pgo\dotnet-pgo.dll
  R2RTest -> C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\R2RTest\R2RTest.dll
  ILCompiler.DependencyAnalysisFramework -> C:\Users\calope\source\repos\runtime\artifacts\bin\ILCompiler.DependencyAnalysisFramework\arm64\Release\ILCompiler.DependencyAnalysisFramework.dll
  ILCompiler.ReadyToRun -> C:\Users\calope\source\repos\runtime\artifacts\bin\ILCompiler.ReadyToRun\arm64\Release\ILCompiler.ReadyToRun.dll
C:\Program Files\dotnet\sdk\6.0.100-preview.2.21155.3\Microsoft.Common.CurrentVersion.targets(4941,5): error MSB3030: Could not copy the file "C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\jitinterface_arm64.dll" because it was not found. [C:\Users\calope\source\repos\runtime\src\coreclr\tools\aot\crossgen2\crossgen2.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.2.21155.3\Microsoft.Common.CurrentVersion.targets(4941,5): error MSB3030: Could not copy the file "C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\clrjit.dll" because it was not found. [C:\Users\calope\source\repos\runtime\src\coreclr\tools\aot\crossgen2\crossgen2.csproj]

Build FAILED.

C:\Program Files\dotnet\sdk\6.0.100-preview.2.21155.3\Microsoft.Common.CurrentVersion.targets(4941,5): error MSB3030: Could not copy the file "C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\jitinterface_arm64.dll" because it was not found. [C:\Users\calope\source\repos\runtime\src\coreclr\tools\aot\crossgen2\crossgen2.csproj]
C:\Program Files\dotnet\sdk\6.0.100-preview.2.21155.3\Microsoft.Common.CurrentVersion.targets(4941,5): error MSB3030: Could not copy the file "C:\Users\calope\source\repos\runtime\artifacts\bin\coreclr\windows.arm64.Release\clrjit.dll" because it was not found. [C:\Users\calope\source\repos\runtime\src\coreclr\tools\aot\crossgen2\crossgen2.csproj]
    0 Warning(s)
    2 Error(s)

Time Elapsed 00:00:51.28
Build failed with exit code 1. Check errors above.
Some builds failed:
        Configuration: Debug, Architecture: arm64
  ```
</details>

### PR DESCRIPTION
**Pull request by adamsitnik**
fixes #50661

tested on my Surface PRO X, works like a charm!

cc @carlossanlop

### CODE REVIEW

**Review by SingleAccretion — COMMENTED** [review]
Just a little nit to use `ARM64` instead of `ARM` as we use `ARM` to mean AArch32 in the runtime.

File: docs/workflow/requirements/windows-requirements.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ Visual Studio 2019 installation process:
   - **.NET Desktop Development** with all default components,
   - **Desktop Development with C++** with all default components.
 - The build tools (CMake, Ninja and Python) can be downloaded and installed separately (see detailed instructions in the [section below](#build-tools)) or by selecting the following **Individual Components**:
-  - **C++ CMake tools for Windows** (includes Ninja),
+  - **C++ CMake tools for Windows** (includes Ninja, but might not work on ARM machines),
```
[the remark this answers is not in the document]
Comment (SingleAccretion) [reply docs/workflow/requirements/windows-requirements.md:29]: But does this actually work on the 32 machines? Windows ARM is a "barely supported" configuration. If it does, no problem then.

**Review by hoyosjs — APPROVED** [review]
Other than @SingleAccretion's comments, I expect this to work - even if it's unsupported. Thanks,

**Review by stephentoub — APPROVED** [review]

**Comment by adamsitnik** [comment]
I've tested it and I was not able to buil `arm` arch, just `arm64`, so I've applied @SingleAccretion suggestions

**Pull request merged**